webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.hooks.js

```diff
@@ -0,0 +1,14 @@
+import { useState, useContext } from 'react';
+import { ModalContext } from './ForemanModal';
+
+export const useModalState = initialModalState => {
+  const [modalOpen, setModalOpen] = useState(initialModalState);
```

Comment:
@amirfefer right, This is just the modal's internal state.  I haven't implemented the modal id's or the connection to Redux yet (that will be the next PR).  Once complete, the modal id and open state will be props supplied to the component from the Redux store.

Reply:
:+1: 
But we need to be careful mixing between inner state and redux, I'm in favor to move it totally to redux

Reply:
@amirfefer I double checked, the only place I am using the custom state hook is the stories and the tests.  When redux is added, it won't conflict with anything inside `<ForemanModal>`.

Reply:
I think when redux will be added there is no point of using the local state, it might confuse other developers, especially when it implemented in the storybook

Reply:
@amirfefer Ok, moving the state hook to the storybook file to make it more clear that that's its only use. :)